numpydoc/cli.py

```diff
@@ -0,0 +1,68 @@
+"""The CLI for numpydoc."""
+
+import argparse
+import ast
+from typing import List, Sequence, Union
+
+from .docscrape_sphinx import get_doc_object
+from .hooks import validate_docstrings
+from .validate import Validator, validate
+
+
+def render_object(import_path: str, config: Union[List[str], None] = None) -> int:
+    """Test numpydoc docstring generation for a given object."""
+    # TODO: Move Validator._load_obj to a better place than validate
+    print(get_doc_object(Validator._load_obj(import_path), config=dict(config or [])))
+    return 0
+
+
+def validate_object(import_path: str) -> int:
+    """Run numpydoc docstring validation for a given object."""
+    exit_status = 0
+    results = validate(import_path)
+    for err_code, err_desc in results["errors"]:
+        exit_status += 1
+        print(":".join([import_path, err_code, err_desc]))
+    return exit_status
+
+
+def main(argv: Union[Sequence[str], None] = None) -> int:
+    """CLI for numpydoc."""
+    ap = argparse.ArgumentParser(prog="numpydoc", description=__doc__)
+    subparsers = ap.add_subparsers(title="subcommands")
+
+    def _parse_config(s):
+        key, _, value = s.partition("=")
+        value = ast.literal_eval(value)
+        return key, value
+
+    render = subparsers.add_parser(
+        "render",
+        description="Test numpydoc docstring generation for a given object.",
+        help="generate the docstring with numpydoc",
+    )
+    render.add_argument("import_path", help="e.g. numpy.ndarray")
+    render.add_argument(
+        "-c",
+        "--config",
+        type=_parse_config,
+        action="append",
+        help="key=val where val will be parsed by literal_eval, "
+        "e.g. -c use_plots=True. Multiple -c can be used.",
+    )
+    render.set_defaults(func=render_object)
+
+    validate = subparsers.add_parser(
+        "validate",
+        description="Validate the docstring with numpydoc.",
```

Comment:
Can we clarify here the distinction between `validate` and `lint`. They do the same, except the one operates on a function and the other on a file? But `validate` doesn't allow disabling certain rules, which confused me.

Reply:
I think it works for non-functions as well. How about this?

> Validate an object's docstring conforms to the numpydoc standard.

Reply:
"Validate that an object's docstring conforms to the numpydoc standard" :+1:

Reply:
Or, "Determine (or check?) whether an object's docstring..."

Reply:
Otherwise just: "Validate an object's docstring against the numpydoc standard"

Reply:
I may prefer the last version, but anything gramatically sound is fine.

Reply:
> I'm not sure what to do about the second part of your comment. If you run `numpydoc validate --help` you see what can be passed as arguments, and there aren't any for disabling rules.

This was more of a question: *should* there be the option of disabling certain rules? Why are these not exactly equivalent in their functionality?

Reply:
> > I'm not sure what to do about the second part of your comment. If you run `numpydoc validate --help` you see what can be passed as arguments, and there aren't any for disabling rules.
> 
> This was more of a question: _should_ there be the option of disabling certain rules? Why are these not exactly equivalent in their functionality?

While I agree this should eventually be the case, this definitely seems out of scope for this PR, but at least the separate subcommands will not draw attention to arguments that have no affect like before (see https://github.com/numpy/numpydoc/issues/459).